### ISSUE TITLE
Period comparison against a timedelta obj raises an error in PyPy

### ISSUE THREAD

**Issue by ohadravid** [sdispater/pendulum#173]
The following code raises an error under PyPy, but works in CPython.

```python
import sys
sys.version
# '3.5.3 (3f6eaa010fce, Jan 11 2018, 05:27:47)\n[PyPy 5.10.1 with GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)]'

import datetime as dt
import pendulum

start_time = pendulum.now()
end_time = pendulum.now()

period_interval = end_time - start_time


period_interval
# <Period [2018-01-23T17:54:24.812611+01:00 -> 2018-01-23T17:54:24.813157+01:00]>

period_interval < dt.timedelta(seconds=6)
```

The error is
```
TypeError
Traceback (most recent call last)
<ipython-input-5-155c77b08c1b> in <module>()
----> 1 period_interval < dt.timedelta(seconds=6)

~/Projects/pypy3-v5.10.1-osx64/lib-python/3/datetime.py in __lt__(self, other)
    596     def __lt__(self, other):
    597         if isinstance(other, timedelta):
--> 598             return self._cmp(other) < 0
    599         else:
    600             _cmperror(self, other)

~/Projects/pypy3-v5.10.1-osx64/lib-python/3/datetime.py in _cmp(self, other)
    614     def _cmp(self, other):
    615         assert isinstance(other, timedelta)
--> 616         return _cmp(self._getstate(), other._getstate())
    617 
    618     def __hash__(self):

~/Projects/pypy3-v5.10.1-osx64/lib-python/3/datetime.py in _cmp(x, y)
      9 
     10 def _cmp(x, y):
---> 11     return 0 if x == y else 1 if x > y else -1
     12 
     13 MINYEAR = 1

TypeError: unorderable types: Pendulum > int
```

While CPython datetime does the comparison in C using days & seconds [_datetimemodule.c#L2104](https://github.com/python/cpython/blob/master/Modules/_datetimemodule.c#L2104), 
PyPy (and the pure python reference) use `timedelta._cmp` [datetime.py#L730](https://github.com/python/cpython/blob/master/Lib/datetime.py#L730) which uses `timedelta._getstate` [datetime.py#L748](https://github.com/python/cpython/blob/master/Lib/datetime.py#L748).

Unfortunately, `Period` implements `_getstate` [period.py#L303](https://github.com/sdispater/pendulum/blob/1.x/pendulum/period.py#L303), but doesn't implement `_cmp`, so the base's `_cmp` is called, and an error is raised.

The fix is to implement `_cmp` for `Period`.

As a workaround, using `period_interval.as_interval() < dt.timedelta(seconds=6)` works on both versions of python.

Thanks!

P.S.
Pendulum is awesome! 
We have a very timezone-sensitive system, and we migrated from Arrow and never looked back :)

**Comment by sdispater** [comment]
Thanks for the kind words :-)

It should be fixed in commit 3a0513c.

It will land in the next bugfix release.

**Comment by sdispater** [comment]
This should be fixed in the latest `1.4.1` release :-)
